File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/Address.cpp
// Address.cpp 
#include "Address.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Address::Address(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //streetAddressNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* streetAddressNode = xmlNode->FirstChildElement("streetAddress");

   if(streetAddressNode){streetAddressIsNull_ = false;}
   else{streetAddressIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- streetAddressNode , address : " << streetAddressNode << std::endl;
   #endif
   if(streetAddressNode)
   {
      if(streetAddressNode->Attribute("href") || streetAddressNode->Attribute("id"))
      {
          if(streetAddressNode->Attribute("id"))
          {
             streetAddressIDRef_ = streetAddressNode->Attribute("id");
             streetAddress_ = boost::shared_ptr<StreetAddress>(new StreetAddress(streetAddressNode));
             streetAddress_->setID(streetAddressIDRef_);
             IDManager::instance().setID(streetAddressIDRef_,streetAddress_);
          }
          else if(streetAddressNode->Attribute("href")) { streetAddressIDRef_ = streetAddressNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { streetAddress_ = boost::shared_ptr<StreetAddress>(new StreetAddress(streetAddressNode));}
   }

   //cityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* cityNode = xmlNode->FirstChildElement("city");

   if(cityNode){cityIsNull_ = false;}
   else{cityIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- cityNode , address : " << cityNode << std::endl;
   #endif
   if(cityNode)
   {
      if(cityNode->Attribute("href") || cityNode->Attribute("id"))
      {
          if(cityNode->Attribute("id"))
          {
             cityIDRef_ = cityNode->Attribute("id");
             city_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(cityNode));
             city_->setID(cityIDRef_);
             IDManager::instance().setID(cityIDRef_,city_);
          }
          else if(cityNode->Attribute("href")) { cityIDRef_ = cityNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { city_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(cityNode));}
   }

   //stateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* stateNode = xmlNode->FirstChildElement("state");

   if(stateNode){stateIsNull_ = false;}
   else{stateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- stateNode , address : " << stateNode << std::endl;
   #endif
   if(stateNode)
   {
      if(stateNode->Attribute("href") || stateNode->Attribute("id"))
      {
          if(stateNode->Attribute("id"))
          {
             stateIDRef_ = stateNode->Attribute("id");
             state_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(stateNode));
             state_->setID(stateIDRef_);
             IDManager::instance().setID(stateIDRef_,state_);
          }
          else if(stateNode->Attribute("href")) { stateIDRef_ = stateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { state_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(stateNode));}
   }

   //countryNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* countryNode = xmlNode->FirstChildElement("country");

   if(countryNode){countryIsNull_ = false;}
   else{countryIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- countryNode , address : " << countryNode << std::endl;
   #endif
   if(countryNode)
   {
      if(countryNode->Attribute("href") || countryNode->Attribute("id"))
      {
          if(countryNode->Attribute("id"))
          {
             countryIDRef_ = countryNode->Attribute("id");
             country_ = boost::shared_ptr<CountryCode>(new CountryCode(countryNode));
             country_->setID(countryIDRef_);
             IDManager::instance().setID(countryIDRef_,country_);
          }
          else if(countryNode->Attribute("href")) { countryIDRef_ = countryNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { country_ = boost::shared_ptr<CountryCode>(new CountryCode(countryNode));}
   }

   //postalCodeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* postalCodeNode = xmlNode->FirstChildElement("postalCode");

   if(postalCodeNode){postalCodeIsNull_ = false;}
   else{postalCodeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- postalCodeNode , address : " << postalCodeNode << std::endl;
   #endif
   if(postalCodeNode)
   {
      if(postalCodeNode->Attribute("href") || postalCodeNode->Attribute("id"))
      {
          if(postalCodeNode->Attribute("id"))
          {
             postalCodeIDRef_ = postalCodeNode->Attribute("id");
             postalCode_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(postalCodeNode));
             postalCode_->setID(postalCodeIDRef_);
             IDManager::instance().setID(postalCodeIDRef_,postalCode_);
          }
          else if(postalCodeNode->Attribute("href")) { postalCodeIDRef_ = postalCodeNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { postalCode_ = boost::shared_ptr<XsdTypeString>(new XsdTypeString(postalCodeNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<StreetAddress> Address::getStreetAddress()
{
   if(!this->streetAddressIsNull_){
        if(streetAddressIDRef_ != ""){
             return boost::shared_static_cast<StreetAddress>(IDManager::instance().getID(streetAddressIDRef_));
        }else{
             return this->streetAddress_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<StreetAddress>();
   }
}
boost::shared_ptr<XsdTypeString> Address::getCity()
{
   if(!this->cityIsNull_){
        if(cityIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeString>(IDManager::instance().getID(cityIDRef_));
        }else{
             return this->city_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeString>();
   }
}
boost::shared_ptr<XsdTypeString> Address::getState()
{
   if(!this->stateIsNull_){
        if(stateIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeString>(IDManager::instance().getID(stateIDRef_));
        }else{
             return this->state_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeString>();
   }
}
boost::shared_ptr<CountryCode> Address::getCountry()
{
   if(!this->countryIsNull_){
        if(countryIDRef_ != ""){
             return boost::shared_static_cast<CountryCode>(IDManager::instance().getID(countryIDRef_));
        }else{
             return this->country_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CountryCode>();
   }
}
boost::shared_ptr<XsdTypeString> Address::getPostalCode()
{
   if(!this->postalCodeIsNull_){
        if(postalCodeIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeString>(IDManager::instance().getID(postalCodeIDRef_));
        }else{
             return this->postalCode_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeString>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/Address.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Address : ISerialized
    {
        public Address(XmlNode xmlNode)
        {
            XmlNode streetAddressNode = xmlNode.SelectSingleNode("streetAddress");
            
            if (streetAddressNode != null)
            {
                if (streetAddressNode.Attributes["href"] != null || streetAddressNode.Attributes["id"] != null) 
                {
                    if (streetAddressNode.Attributes["id"] != null) 
                    {
                        streetAddressIDRef_ = streetAddressNode.Attributes["id"].Value;
                        StreetAddress ob = new StreetAddress(streetAddressNode);
                        IDManager.SetID(streetAddressIDRef_, ob);
                    }
                    else if (streetAddressNode.Attributes["href"] != null)
                    {
                        streetAddressIDRef_ = streetAddressNode.Attributes["href"].Value;
                    }
                    else
                    {
                        streetAddress_ = new StreetAddress(streetAddressNode);
                    }
                }
                else
                {
                    streetAddress_ = new StreetAddress(streetAddressNode);
                }
            }
            
        
            XmlNode cityNode = xmlNode.SelectSingleNode("city");
            
            if (cityNode != null)
            {
                if (cityNode.Attributes["href"] != null || cityNode.Attributes["id"] != null) 
                {
                    if (cityNode.Attributes["id"] != null) 
                    {
                        cityIDRef_ = cityNode.Attributes["id"].Value;
                        XsdTypeString ob = new XsdTypeString(cityNode);
                        IDManager.SetID(cityIDRef_, ob);
                    }
                    else if (cityNode.Attributes["href"] != null)
                    {
                        cityIDRef_ = cityNode.Attributes["href"].Value;
                    }
                    else
                    {
                        city_ = new XsdTypeString(cityNode);
                    }
                }
                else
                {
                    city_ = new XsdTypeString(cityNode);
                }
            }
            
        
            XmlNode stateNode = xmlNode.SelectSingleNode("state");
            
            if (stateNode != null)
            {
                if (stateNode.Attributes["href"] != null || stateNode.Attributes["id"] != null) 
                {
                    if (stateNode.Attributes["id"] != null) 
                    {
                        stateIDRef_ = stateNode.Attributes["id"].Value;
                        XsdTypeString ob = new XsdTypeString(stateNode);
                        IDManager.SetID(stateIDRef_, ob);
                    }
                    else if (stateNode.Attributes["href"] != null)
                    {
                        stateIDRef_ = stateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        state_ = new XsdTypeString(stateNode);
                    }
                }
                else
                {
                    state_ = new XsdTypeString(stateNode);
                }
            }
            
        
            XmlNode countryNode = xmlNode.SelectSingleNode("country");
            
            if (countryNode != null)
            {
                if (countryNode.Attributes["href"] != null || countryNode.Attributes["id"] != null) 
                {
                    if (countryNode.Attributes["id"] != null) 
                    {
                        countryIDRef_ = countryNode.Attributes["id"].Value;
                        CountryCode ob = new CountryCode(countryNode);
                        IDManager.SetID(countryIDRef_, ob);
                    }
                    else if (countryNode.Attributes["href"] != null)
                    {
                        countryIDRef_ = countryNode.Attributes["href"].Value;
                    }
                    else
                    {
                        country_ = new CountryCode(countryNode);
                    }
                }
                else
                {
                    country_ = new CountryCode(countryNode);
                }
            }
            
        
            XmlNode postalCodeNode = xmlNode.SelectSingleNode("postalCode");
            
            if (postalCodeNode != null)
            {
                if (postalCodeNode.Attributes["href"] != null || postalCodeNode.Attributes["id"] != null) 
                {
                    if (postalCodeNode.Attributes["id"] != null) 
                    {
                        postalCodeIDRef_ = postalCodeNode.Attributes["id"].Value;
                        XsdTypeString ob = new XsdTypeString(postalCodeNode);
                        IDManager.SetID(postalCodeIDRef_, ob);
                    }
                    else if (postalCodeNode.Attributes["href"] != null)
                    {
                        postalCodeIDRef_ = postalCodeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        postalCode_ = new XsdTypeString(postalCodeNode);
                    }
                }
                else
                {
                    postalCode_ = new XsdTypeString(postalCodeNode);
                }
            }
            
        
        }
        
    
        #region StreetAddress_
        private StreetAddress streetAddress_;
        public StreetAddress StreetAddress_
        {
            get
            {
                if (this.streetAddress_ != null)
                {
                    return this.streetAddress_; 
                }
                else if (this.streetAddressIDRef_ != null)
                {
                    streetAddress_ = IDManager.getID(streetAddressIDRef_) as StreetAddress;
                    return this.streetAddress_; 
                }
                else
                {
                      return this.streetAddress_; 
                }
            }
            set
            {
                if (this.streetAddress_ != value)
                {
                    this.streetAddress_ = value;
                }
            }
        }
        #endregion
        
        public string streetAddressIDRef_ { get; set; }
        #region City_
        private XsdTypeString city_;
        public XsdTypeString City_
        {
            get
            {
                if (this.city_ != null)
                {
                    return this.city_; 
                }
                else if (this.cityIDRef_ != null)
                {
                    city_ = IDManager.getID(cityIDRef_) as XsdTypeString;
                    return this.city_; 
                }
                else
                {
                      return this.city_; 
                }
            }
            set
            {
                if (this.city_ != value)
                {
                    this.city_ = value;
                }
            }
        }
        #endregion
        
        public string cityIDRef_ { get; set; }
        #region State_
        private XsdTypeString state_;
        public XsdTypeString State_
        {
            get
            {
                if (this.state_ != null)
                {
                    return this.state_; 
                }
                else if (this.stateIDRef_ != null)
                {
                    state_ = IDManager.getID(stateIDRef_) as XsdTypeString;
                    return this.state_; 
                }
                else
                {
                      return this.state_; 
                }
            }
            set
            {
                if (this.state_ != value)
                {
                    this.state_ = value;
                }
            }
        }
        #endregion
        
        public string stateIDRef_ { get; set; }
        #region Country_
        private CountryCode country_;
        public CountryCode Country_
        {
            get
            {
                if (this.country_ != null)
                {
                    return this.country_; 
                }
                else if (this.countryIDRef_ != null)
                {
                    country_ = IDManager.getID(countryIDRef_) as CountryCode;
                    return this.country_; 
                }
                else
                {
                      return this.country_; 
                }
            }
            set
            {
                if (this.country_ != value)
                {
                    this.country_ = value;
                }
            }
        }
        #endregion
        
        public string countryIDRef_ { get; set; }
        #region PostalCode_
        private XsdTypeString postalCode_;
        public XsdTypeString PostalCode_
        {
            get
            {
                if (this.postalCode_ != null)
                {
                    return this.postalCode_; 
                }
                else if (this.postalCodeIDRef_ != null)
                {
                    postalCode_ = IDManager.getID(postalCodeIDRef_) as XsdTypeString;
                    return this.postalCode_; 
                }
                else
                {
                      return this.postalCode_; 
                }
            }
            set
            {
                if (this.postalCode_ != value)
                {
                    this.postalCode_ = value;
                }
            }
        }
        #endregion
        
        public string postalCodeIDRef_ { get; set; }
        
    
        
    
    }
    
}

